pages/_app.tsx

```diff
@@ -135,16 +135,40 @@ export default function PlanetWeb({ Component, pageProps, err }: any) {
 
   const [showVideo, setshowVideo] = React.useState(true);
 
+  // if localShowVideo is undefined
+  // set localShowVideo is true and show the video
+  // if localShowVideo is true show the video
+  // if localShowVideo is false hide the video
+
+  const [localShowVideo, setLocalShowVideo] = React.useState(false);
+
   React.useEffect(() => {
-    if (typeof window !== 'undefined') {
-      if (localStorage.getItem('hidePreview')) {
-        setshowVideo(false);
-      }
-      if (router.pathname !== '/') {
-        setshowVideo(false);
+    if (router.pathname === '/') {
+      if (typeof window !== 'undefined') {
+        if (localStorage.getItem('showVideo')) {
+          if (localStorage.getItem('showVideo') === 'true') {
+            setLocalShowVideo(true);
+          } else {
+            setLocalShowVideo(false);
+          }
```

Comment:
You can directly do `setLocalShowVideo(localStorage.getItem('showVideo') === 'true')`, no need for conditional loops

Reply:
local storage values are returned as string.